### ISSUE TITLE
ExcludeFromBlockingCallAnalysisAttribute not working as expected

### ISSUE THREAD

**Issue by robv8r** [meziantou/Meziantou.Analyzer#1152]
### Version of the Meziantou.Analyzer NuGet package

3.0.80

### Rule Identifier

MA0042

### Target Framework

net10.0

### C# Language version

C# 14.0

### Description

Per [MA0042.md](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0042.md#configuration) and the [README.md for Meziantou.Analyzer.Annotations](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer.Annotations/README.md) , adding the following assembly-level attribute should suppress MA0042 for `DbConnection.CreateCommand()` but it doesn't.

```c#
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("M:System.Data.Common.DbConnection.CreateCommand")]
```

### Reproduction Steps

Minimal code:

```c#
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("M:System.Data.Common.DbConnection.CreateCommand")]

namespace Example;

using System.Data.Common;
using System.Diagnostics.CodeAnalysis;
using Microsoft.Data.SqlClient;

internal sealed class DataService(IConnectionFactory connectionFactory)
{
    private readonly IConnectionFactory connectionFactory = connectionFactory ?? throw new ArgumentNullException(nameof(connectionFactory));

    public async Task<int> GetDataAsync(CancellationToken cancellationToken = default)
    {
        var conn = this.connectionFactory.CreateConnection();
        await using (conn.ConfigureAwait(false))
        {
            await conn.OpenAsync(cancellationToken).ConfigureAwait(false);
            using var cmd = conn.CreateCommand(); // 👈 MA0042 is reported, but shouldn't be.
            cmd.CommandText = "SELECT 1";
            var reader = await cmd.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
            await using (reader.ConfigureAwait(false))
            {
                if (await reader.ReadAsync(cancellationToken).ConfigureAwait(false))
                {
                    return reader.GetInt32(0);
                }
            }
        }

        throw new InvalidOperationException("No data returned");
    }
}

internal interface IConnectionFactory
{
    DbConnection CreateConnection();
}

internal sealed class SqlServerConnectionFactory(string connectionString) : IConnectionFactory
{
    private readonly string connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));

    public DbConnection CreateConnection() => new SqlConnection(this.connectionString);
}
```

### Other information

The `Example.csproj` file looks like this:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net10.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <AnalysisLevel>latest-all</AnalysisLevel>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Meziantou.Analyzer" Version="3.0.80">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
    <PackageReference Include="Meziantou.Analyzer.Annotations" Version="1.3.0" />
    <PackageReference Include="Microsoft.Data.SqlClient" Version="7.0.1" />
  </ItemGroup>

</Project>
```

I've tested several different approaches, but none suppress MA0042 for `DbConnection.CreateCommand()`:

```c#
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysisAttribute(typeof(System.Data.Common.DbConnection), "CreateCommand")]
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysisAttribute(typeof(System.Data.IDbConnection), "CreateCommand")]

[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("M:System.Data.Common.DbConnection.CreateCommand()")]
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("M:System.Data.IDbConnection.CreateCommand()")]

[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("M:System.Data.Common.DbConnection.CreateCommand")]
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("M:System.Data.IDbConnection.CreateCommand")]

[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("~M:System.Data.Common.DbConnection.CreateCommand()~System.Data.Common.DbCommand")]
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("~M:System.Data.IDbConnection.CreateCommand()~System.Data.IDbCommand")]

[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("System.Data.Common.DbConnection.CreateCommand")]
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("System.Data.IDbConnection.CreateCommand")]
```

**Comment by robv8r** [comment]
I've also tried the following attribute:

```c#
[assembly: Meziantou.Analyzer.Annotations.ExcludeFromBlockingCallAnalysis("~M:System.Data.Common.DbConnection.CreateCommand~System.Data.Common.DbCommand")]
```

It's similar to one of the above example attributes, but doesn't include parenthesis after the method name.

Aside: Could this be related to methods that do not return `void`?

**Comment by meziantou** [comment]
The attribute is for the method and it's async equivalent at the moment (Sample vs SampleAsync). It doesn't support `using` vs `await using`. I'm still not sure what's the correct way to handle it. I need to experiment a bit.

**Comment by robv8r** [comment]
Thank you VERY much for looking into this.

I'm very impressed by the Meziantou.Analyzer and your quick resolutions to submitted issues and requests.

**Comment by meziantou** [comment]
There will be a new attribute NonAwaitableTypeAttribute to declare types that doesn't need to be awaited.
That name can change in the future. Still not very sure if it is the best way to do it.

In fact, I wonder if there should be NonAwaitableTypeAttribute and NonAsyncDisposableAttribute to better differentiate the exclusion.

**Comment by meziantou** [comment]
So, I've added the NonAsyncDisposableAttribute for the `await using` case

**Comment by robv8r** [comment]
`NonAsyncDisposableAttribute` seems to be exactly what I needed.  Thank you very much for resolving this.
